cdap-authorization-dataset-plugin/src/main/java/co/cask/cdap/security/authorization/DatasetBasedAuthorizer.java

```diff
@@ -144,4 +148,39 @@ public void apply(ACLDataset acls) throws Exception {
       }
     }, acls.get());
   }
+
+  @Override
+  public void createRole(Role role) throws RoleAlreadyExistsException {
+    throw new UnsupportedOperationException("Role based operation is not supported.");
+  }
+
+  @Override
+  public void dropRole(Role role) throws RoleNotFoundException {
+    throw new UnsupportedOperationException("Role based operation is not supported.");
+  }
+
+  @Override
+  public void addRoleToPrincipal(Role role, Principal principal) throws RoleNotFoundException {
+    throw new UnsupportedOperationException("Role based operation is not supported.");
+  }
+
+  @Override
+  public void removeRoleFromPrincipal(Role role, Principal principal) throws RoleNotFoundException {
+    throw new UnsupportedOperationException("Role based operation is not supported.");
+  }
+
+  @Override
+  public Set<Role> listRoles(Principal principal) {
+    return null;
```

Comment:
if we do decide to not support this, then we should not return null, but throw an ``UnsupportedOperationException``